app/api/business/validators/team_validator.py

```diff
@@ -0,0 +1,77 @@
+import collections
+import pendulum
+import re
+from flask import current_app
+from app.api.services import team_service
+
+
+class TeamValidator(object):
+    def __init__(self, team):
+        self.team = team
+
+    def validate_all(self):
+        result = (
+            self.validate_basics() +
+            self.validate_team_members()
+        )
+        warnings = [n for n in result if n.get('severity', 'error') == 'warning']
+        errors = [n for n in result if n.get('severity', 'error') == 'error']
+        validation_result = collections.namedtuple('Notification', ['warnings', 'errors'])
+        return validation_result(warnings=warnings, errors=errors)
+
+    def validate_basics(self):
+        errors = []
+        if not self.team.name or not self.team.name.replace(' ', ''):
+            errors.append({
+                'message': 'Team name is required.',
+                'severity': 'error',
+                'step': 'about',
+                'id': 'T001'
+            })
+
+        if self.team.email_address:
+            if '@' not in self.team.email_address or self.team.email_address.count('@') > 1:
+                errors.append({
+                    'message': 'Team email is not valid',
+                    'severity': 'error',
+                    'step': 'about',
+                    'id': 'T002'
+                })
+
+        return errors
+
+    def validate_team_members(self):
```

Comment:
This is not currently validating whether the `team_members` contains a non buyer member (it is possible to manipulate the POST to `/api/2/team/<int:team_id>/update` to include user ids of non buyer users).

Reply:
add validation